### DISCUSSION TITLE
Composing tailwind components does not work as expected

### DISCUSSION THREAD

**Discussion by herrKlein** [tailwindlabs/tailwindcss#11132] [Help] [answered]
Hi Y'all. I'm deep into tailwind stylesheets for theming a webcomponent lib.

I can't get this to work, can someone explain why the class .base-red does not deliver a button with a red background? https://play.tailwindcss.com/giTTntJ87a?file=css

```css
@layer components {
  .base {
    @apply cursor-pointer rounded-md px-2 py-1 text-gray-900
    outline-none ring-1 ring-inset ring-gray-300;
  }
  .base.red {
    @apply bg-red-300;
  }
}
.base-red {
  @apply base red;
}
```

```html
<button class="base">.base </button>
<button class="base red">.base .red</button>
<button class="base-red">.base-red</button>
```

**Comment by wongjn** [comment, marked as answer]
There is no standalone `.red` rule. I assume you were hoping that the CSS properties in the `.base.red` rule would be extracted into the `.base-red` rule. I don't believe that `@apply` was really meant to be used in such a way and is more to be used as a "simple" escape hatch for when you can't use Tailwind CSS class names on an element directly.

**Reply by herrKlein** [reply]
thanks for the quick response, I tried several other approaches but none seem te work..
